pkgs/development/libraries/ffmpeg/generic.nix

```diff
@@ -539,7 +539,7 @@ stdenv.mkDerivation (finalAttrs: {
   # TODO This was always in buildInputs before, why?
   buildInputs = optionals withFullDeps [ libdc1394 ]
   ++ optionals (withFullDeps && !stdenv.isDarwin) [ libraw1394 ] # TODO where does this belong to
-  ++ optionals (withNvdec || withNvenc) [ nv-codec-headers ]
+  ++ optionals (withNvdec || withNvenc) [ nv-codec-headers-11 ]
```

Comment:
I think I'd prefer to keep the older version in at least ffmpeg_4 and perhaps ffmpeg_5 for now. Please guard this change behind a `lib.versionAtLeast version "6.0"`.

Reply:
I'm now taking both nv-codec-headers and nv-codec-headers-11 as arguments, and only use the -11 variant on 6+, which keeps ffmpeg-4/5 untouched.